plugins/org.eclipse.reddeer.swt/src/org/eclipse/reddeer/swt/condition/NewShellIsOpenedOrIsNotAvailable.java

```diff
@@ -0,0 +1,56 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.eclipse.reddeer.swt.condition;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.reddeer.common.condition.AbstractWaitCondition;
+import org.eclipse.reddeer.core.lookup.ShellLookup;
+import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
+import org.eclipse.swt.widgets.Shell;
+
+public class NewShellIsOpenedOrIsNotAvailable extends AbstractWaitCondition {
```

Comment:
Please add java doc because I really have no idea the name 'NewShellIsOpenedOrIsNotAvailable' is a bit confusing.